tests/cupy_tests/math_tests/test_matmul.py

```diff
@@ -0,0 +1,48 @@
+import operator
+import unittest
+
+import numpy
+
+from cupy import testing
+
+
+@testing.parameterize(
+    *testing.product({
+        'arrays': [
+            (numpy.random.randn(*s1), numpy.random.randn(*s2)) for s1, s2 in [
+                ((3, 2), (2, 4)),
+                ((2,), (2, 4)),
+                ((3, 2), (2,)),
+                ((2,), (2,)),
+                ((5, 3, 2), (5, 2, 4)),
+                ((5, 3, 2), (2, 4)),
+                ((3, 2), (5, 2, 4)),
+                ((5, 3, 2), (1, 2, 4)),
+                ((1, 3, 2), (5, 2, 4)),
+                ((5, 3, 2), (2,)),
+                ((2,), (5, 2, 4)),
+                ((6, 5, 3, 2), (6, 5, 2, 4)),
+                ((6, 5, 3, 2), (6, 1, 2, 4)),
+                ((6, 5, 3, 2), (1, 5, 2, 4)),
+                ((6, 5, 3, 2), (1, 1, 2, 4)),
+                ((6, 1, 3, 2), (6, 5, 2, 4)),
+                ((1, 5, 3, 2), (6, 5, 2, 4)),
+                ((1, 1, 3, 2), (6, 5, 2, 4)),
+                ((3, 2), (6, 5, 2, 4)),
+                ((6, 5, 3, 2), (2, 4)),
+                ((2,), (6, 5, 2, 4)),
+                ((6, 5, 3, 2), (2,)),
+            ]],
+    }))
+@testing.gpu
+class TestMatmul(unittest.TestCase):
+
+    # _multiprocess_can_split_ = True
+
+    @testing.for_all_dtypes(name='dtype1')
+    @testing.for_all_dtypes(name='dtype2')
+    @testing.numpy_cupy_array_almost_equal()
+    def test_matmul(self, xp, dtype1, dtype2):
```

Comment:
`matmul` operator is not supported on py2.
see also 
https://github.com/pfnet/chainer/blob/bbddcc46b67c49f9e09dcedf40260d7241964cd0/tests/chainer_tests/functions_tests/math_tests/test_basic_math.py#L1059